snare/cloner.py

```diff
@@ -155,16 +165,26 @@ async def get_body(self, session):
             file_name, hash_name = self._make_filename(current_url)
             self.logger.debug("Cloned file: %s", file_name)
             data = None
+            headers = []
             content_type = None
-            try:
-                response = await session.get(current_url, headers={"Accept": "text/html"}, timeout=10.0)
-                headers = self.get_headers(response)
-                content_type = response.content_type
-                data = await response.read()
-            except (aiohttp.ClientError, asyncio.TimeoutError) as client_error:
-                self.logger.error(client_error)
-            else:
-                await response.release()
+            if page:
```

Comment:
Do you think it will be a good idea to split the headless/non-headless parts into the separate functions (or even classes with the common interface)?